cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -262,8 +262,19 @@ func ValidateCloudProvider(provider string, fldPath *field.Path) field.ErrorList
 }
 
 func ValidateMixedArguments(flag *pflag.FlagSet) error {
-	if flag.Changed("config") && flag.NFlag() != 1 {
-		return fmt.Errorf("can not mix '--config' with other arguments")
+	if flag.Changed("config") {
+		numFlagsAllowedToChange := 1
+		flagsAllowedToChangeWithConfig := []string{"skip-preflight-checks", "skip-token-print"}
```

Comment:
You probably want Flagset.Visit https://godoc.org/github.com/spf13/pflag#FlagSet.Visit

```go
mixedFlags := []string{}
flag.Visit(func(f &pflag.Flag) {
    if f.Name == "config" || strings.HasPrefix(f.Name, "skip-") {
        // "skip=*" flags can be set with --config
        continue
    }
    mixedFlags = append(mixedFlags, f.Name)
})
```